docs/architecture/blazor-for-web-forms-developers/state-management.md

```diff
@@ -1,22 +1,98 @@
+
+The signature of the .NET Framework `Session` object is not the same as the ASP.NET Core `Session` object.  Consider [the documentation for the new ASP.NET Core Session](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.http.isession?view=aspnetcore-3.1) before deciding to migrate and use the new session state feature.
+
+Performance considerations and recommendations for application architecture in larger applications with Blazor Server can be found in the [Secure ASP.NET Core Blazor Server Apps article](https://docs.microsoft.com/aspnet/core/security/blazor/server?view=aspnetcore-3.1)
+
+## Maintaining state with Session
+
+Web Forms developers could maintain information about the currently acting user with the `Session` dictionary object.  It is easy enough to add an object with a string key to the `Session` and that object would be available at a later time during the user's interactions with the application.  In the attempt to eliminate managing interacting with HTTP, the Session object made it easy to maintain state.
+
+Session is available in ASP.NET Core and Blazor Server, but is discouraged from use in favor of storing data in a data repository appropriately.  Session state is also not functional when GDPR support is activated and your visitors decline HTTP cookies in your application.
+
+Configuration for ASP.NET Core and Session state is available in the [Session and state management in ASP.NET Core article](https://docs.microsoft.com/aspnet/core/fundamentals/app-state?view=aspnetcore-3.1)
+
+## Application State
+
+The `Application` object in the Web Forms framework provides a massive cross-request repository for interacting with application-scope configuration and state.  Application state was an ideal place to store various application configuration properties that would be referenced by all requests, regardless of the user making the request.  The problem with the `Application` object was that data did not persist across multiple servers and you would lose the state of the application object between restarts.
+
+As with `Session`, it is recommended that data move to a persistent backing store that could be accessed by multiple server instances.  If there is volatile data that you would like to be able to access across requests and users, you could easily store it in a static property or in a singleton object that can be injected into components that require this information or interaction.
+
+The construction of a simple object to maintain application state and its consumption could be as simple as the following implementation:
+
+```csharp
+public class MyApplicationState
+{
+
+    public int VisitorCounter { get; private set; } = 0;
+
+    public void IncrementCounter()
+    {
+
+        VisitorCounter += 1;
+
+    }
+
+}
+```
+
+```csharp
+
+    app.AddSingleton<MyApplicationState>();
+
+```
```

Comment:
```suggestion
```csharp
app.AddSingleton<MyApplicationState>();
```
```

Reply:
You might want to add some context to these C# snippets. Usually, it is a docs-smell (like code smell, but for docs) when you see stacked code blocks like this without text in between.

Reply:
I agree. At a minimum, there should be text pointing out this code goes in `Startup.Configure`.